src/libs/modulegroups.c

```diff
@@ -764,6 +751,25 @@ static gchar *_preset_retrieve_old_layout_updated()
 {
   gchar *ret = NULL;
 
+  // show the search box ?
+  const gchar *show_text_entry = dt_conf_get_string("plugins/darkroom/search_iop_by_text");
+  if(strcmp(show_text_entry, "show search text") == 0)
```

Comment:
May I suggest that this code be in a function as duplicated just below. And the routine should return a dt_lib_modulegroup_iop_visibility_type_t.